executor/src/witgen/jit/effect.rs

```diff
@@ -46,7 +44,8 @@ impl<T: FieldElement, V: Hash + Eq> Effect<T, V> {
                     .chain(vec1.iter().flat_map(|effect| effect.referenced_variables()))
                     .unique(),
             ),
-        }
+        };
+        iter.unique()
```

Comment:
Cool! You could remove a few more `unique()` calls above then..

Reply:
Hm, weird merge problem it seems.